### PR TITLE
Add hooks (pre commit and push)

### LINKED ISSUE #115: Start using precommits
I've seen that we often forget running the linting checker/black before doing commits. Maybe a precommit hook might help us for that? I wanna look into setting this up, but I believe it's client side only. What's your opinion on this? 

Reference: https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/

### PR DESCRIPTION
**Pull request by bh2smith**
Closes #115 

We add `pre-commit` and `pre-push` hook files to be copied into `.git/hooks`. 

To install hooks (this is also added in readme).

```shell
cp hooks/pre-commit .git/hooks/ && chmod +x .git/hooks/pre-commit
cp hooks/pre-push .git/hooks/ && chmod +x .git/hooks/pre-push
```

## Test Plan

I have tested these out myself. You can too.

### CODE REVIEW

**Review by gentrexha — APPROVED** [review]
Works like a charm!

Thank you Ben! It's really nice to have these!

**Comment by bh2smith** [comment]
I think we also need to start skipping the slackbot test. Its not very nice sending out the test messages to actual slack channels. At least until we mock it. Look at all these accidental posts to the multisig channel (because I manually ran the payout this week):

https://cowservices.slack.com/archives/C037UV49JLR/p1666897177728069

**Comment by gentrexha** [comment]
> I think we also need to start skipping the slackbot test. Its not very nice sending out the test messages to actual slack channels. At least until we mock it. Look at all these accidental posts to the multisig channel (because I manually ran the payout this week):
> 
> https://cowservices.slack.com/archives/C037UV49JLR/p1666897177728069

I agree it's not very good to have to spam the actual slack channel. Mocking is definitely a must have in the near future.

File: tests/e2e/test_slack_post.py [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
 import certifi
+import pytest
 from slack import WebClient
 from slack.errors import SlackApiError
 
 from src.fetch.transfer_file import post_to_slack
 
 
 class TestSlackPost(unittest.TestCase):
+    @pytest.mark.skip(reason="Too many accidental slack posts - need to mock this!")
```
Comment (gentrexha) [inline tests/e2e/test_slack_post.py]: A nice to have would be to link to the issue which implements this so we know there's already some info about it.

Comment (bh2smith) [reply tests/e2e/test_slack_post.py]: Sure thing. Will add.

**Pull request merged**